### DISCUSSION TITLE
Feature request: `title-*` class

### DISCUSSION THREAD

**Discussion by alexburner** [saadeghi/daisyui#1742] [Ideas / feature requests]
I'm just getting started using daisyUI, and mostly loving it so far.

However, one thing I'm missing is something like a `title-*` class.

There's the `prose` class, which nicely formats header elements within it... but it's intended for "vanilla HTML you don’t control", and doesn't play well with other daisyUI components (related issue: https://github.com/saadeghi/daisyui/issues/1229)

I'm picturing something like [Bulma's `title is-*` and `subtitle is-*` classes](https://bulma.io/documentation/elements/title/), as a way to get consistent titles within a web app:

<img width="1180" alt="Screenshot 2023-04-08 at 11 18 49 AM" src="https://user-images.githubusercontent.com/409114/230736917-1e04f170-59b6-46de-9d67-2fec3839a95d.png">

As it is, I'm currently wrapping individual header elements with the `prose` class, which feels silly 😅

**Comment by saadeghi** [comment]
It's not just heading, daisyUI doesn't provide any typography styles.  
If we add styles for headings, then we have to provide styles for paragraphs, lists (ul, ol) and everything else `@tailwindcss/typography` is offering. I don't want to enter that area because I think that plugin is doing a good job about handling typography.

I know it doesn't feel right to wrap 1 heading in a `prose` class but another option is using `text-*` and `font-*` utility classes to make text bigger or bolder. I usually do this myself if the whole section is not a `prose`

**Reply by alexburner** [reply]
Thanks for your quick response!

> If we add styles for headings, then we have to provide styles for paragraphs, lists (ul, ol) and everything else

I don't believe that logic is strictly correct — in my experience, HTML text content elements like `p`, `ul`, `ol` are more rare for building web apps, they're usually found in cases where `prose` is appropriate (such as a block of rich text content).

But all web apps do use titles. And it's nice to keep those titles consistent across sections/pages. It's a common "component", even if it's not a complex one.

Like in the Bulma example, they offer `title`, but nothing for lists or otherwise. [They have `content` for those](https://bulma.io/documentation/elements/content/) (similar to `prose`).

> another option is using `text-*` and `font-*` utility classes to make text bigger or bolder

It's true, I could use `text-*` and `font-*` classes to make my own. The same's true with anything in daisyUI, it can all be made with lower-level primitives. But it's nice to have a collection of convenience components that all play well together.

All that being said, I don't think this is a burning priority — I just found myself reaching for it and saw it missing.

**Reply by saadeghi** [reply]
> But it's nice to have a collection of convenience components that all play well together.

In my experience, headings in a real-life app do not always share the same size and boldness pattern unless it's a markdown or article block.  
For example, imagine a landing page, you may need `h1` and `h2` with the same size. Or In an admin panel design you may need a `h3` larger than `h2`. It all depends on the design.  
So I think unless it's markdown block or an article (which can be styled using `prose`), it's better to use `text-*` and `font-*` for headings elsewhere.  
Even if this customizability wasn't the case, merging only 2 class name in 1 doesn't really look efficient 😁
